pkg/adaptor/mongodb.go

```diff
@@ -87,8 +90,32 @@ func NewMongodb(p *pipe.Pipe, path string, extra Config) (StopStartListener, err
 		return m, err
 	}
 
-	m.mongoSession, err = mgo.Dial(m.uri)
+	dialInfo, err := mgo.ParseURL(m.uri)
+	if err != nil {
+		return m, fmt.Errorf("unable to parse uri (%s), %s\n", m.uri, err.Error())
+	}
 
+	if conf.Ssl != nil {
+		tlsConfig := &tls.Config{}
+		if len(conf.Ssl.CaCerts) > 0 {
+			roots := x509.NewCertPool()
+			for _, caCert := range conf.Ssl.CaCerts {
+				ok := roots.AppendCertsFromPEM([]byte(caCert))
+				if !ok {
+					return m, fmt.Errorf("failed to parse root certificate")
+				}
+			}
+			tlsConfig.RootCAs = roots
+		} else {
+			tlsConfig.InsecureSkipVerify = true
+		}
+		dialInfo.DialServer = func(addr *mgo.ServerAddr) (net.Conn, error) {
+			conn, err := tls.Dial("tcp", addr.String(), tlsConfig)
+			return conn, err
```

Comment:
do we want 'direct' as a config option? it's used in seed quite a bit
http://godoc.org/labix.org/v2/mgo#DialInfo